### DISCUSSION TITLE
"Native" Styling

### DISCUSSION THREAD

**Discussion by jrmoulton** [lapce/floem#670] [Ideas]
I've been thinking about how to do "native" looking styles with floem. 

I'm thinking about introducing a simple `NativeStyle` trait or maybe just a method on `View` that is

```rust
pub enum OperatingSystem {
    MacOS,
    Windows,
    Linux
}
/// return a style for each operating system according to whatever the latest style guide from the respective OS is.
fn native_style(os: OperatingSystem) -> Option<Style> {
    None // by default if in `View`
}
```

then each widget would be responsible for providing it's own styles that match the system. 

Then the floem default theme can simply combine the styles that are returned and apply a few Floem specific highlights.

**Comment by dzhou121** [comment]
How about each view is just responsible to provide `fn view_style(&self) -> Option<Style>` out of the box? Since there's no "default Floem style" at the moment, we can make the system "native" style as the default.

**Reply by jrmoulton** [reply]
oh yeah I guess all we really need to do is add some docs to the view_style widget that state that the expectation is that the returned style is native to the platform. 

I was thinking of another method because I wasn't sure if I wanted it to always be applied or if it would be optional but I guess that really is a separate issue.
